osu.Game.Rulesets.Osu.Tests/TestSceneObjectOrderedHitPolicy.cs

```diff
@@ -0,0 +1,491 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+using osu.Framework.Extensions.TypeExtensions;
+using osu.Framework.Screens;
+using osu.Framework.Utils;
+using osu.Game.Beatmaps;
+using osu.Game.Beatmaps.ControlPoints;
+using osu.Game.Replays;
+using osu.Game.Rulesets.Judgements;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Objects.Types;
+using osu.Game.Rulesets.Osu.Mods;
+using osu.Game.Rulesets.Osu.Objects;
+using osu.Game.Rulesets.Osu.Replays;
+using osu.Game.Rulesets.Replays;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Scoring;
+using osu.Game.Screens.Play;
+using osu.Game.Tests.Visual;
+using osuTK;
+
+namespace osu.Game.Rulesets.Osu.Tests
+{
+    public class TestSceneObjectOrderedHitPolicy : RateAdjustedBeatmapTestScene
```

Comment:
I can extract this to a separate PR if you'd like - the implementation of `ObjectOrderedHitPolicy`. This file is basically a copy-paste of `TestSceneStartTimeOrderedHitPolicy` with judgements changed and a new method `TestClickSecondCircleAfterFirstCircleTimeWithFirstCircleJudged`.